core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala

```diff
@@ -52,6 +51,7 @@ abstract class ContextCleanerSuiteBase(val shuffleManager: Class[_] = classOf[Ha
     .setAppName("ContextCleanerSuite")
     .set("spark.cleaner.referenceTracking.blocking", "true")
     .set("spark.cleaner.referenceTracking.blocking.shuffle", "true")
+    .set("spark.cleaner.referenceTracking.cleanCheckpoints", "true")
```

Comment:
I dont like the fact that the this PR touches a configuration that may affect all the tests in an unrelated testsuite. May be it is fine that there is not regression, but then one has to go through all the tests in this TestSuite to be sure.